cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -871,8 +871,7 @@ private void deleteMeta(Table table, Id.Artifact artifactId, byte[] oldData) thr
     // delete the old jar file
 
     try {
-      new NamespacedImpersonator(artifactId.getNamespace().toEntityId(),
-                                 impersonator).impersonate(new Callable<Void>() {
+      new EntityImpersonator(artifactId.getNamespace().toEntityId(), impersonator).impersonate(new Callable<Void>() {
```

Comment:
Artifacts will be owned by the namespace owner so the impersonator takes the namespaceId. Also artifactId is of type Id.Artifact, so there is no artifactId.getNamespaceId()